network/endpoint_windows.go

```diff
@@ -212,17 +215,24 @@ func (nw *network) configureHcnEndpoint(epInfo *EndpointInfo) (*hcn.HostComputeE
 		Name:               infraEpName,
 		HostComputeNetwork: nw.HnsId,
 		Dns: hcn.Dns{
-			Search:     strings.Split(epInfo.DNS.Suffix, ","),
-			ServerList: epInfo.DNS.Servers,
-			Options:    epInfo.DNS.Options,
+			Search:     strings.Split(epInfo.EndpointDNS.Suffix, ","),
+			ServerList: epInfo.EndpointDNS.Servers,
+			Options:    epInfo.EndpointDNS.Options,
 		},
 		SchemaVersion: hcn.SchemaVersion{
 			Major: hcnSchemaVersionMajor,
 			Minor: hcnSchemaVersionMinor,
 		},
-		MacAddress: epInfo.MacAddress.String(),
 	}
 
+	// macAddress type for InfraNIC is like "60:45:bd:12:45:65"
+	macAddress := epInfo.MacAddress.String()
+	if epInfo.NICType != cns.InfraNIC {
+		// convert the format of macAddress that HNS can accept, i.e, "60-45-bd-12-45-65" if NIC type is delegated NIC
+		macAddress = strings.Join(strings.Split(macAddress, ":"), "-")
```

Comment:
Curious to know, is this something new ? When we read the HNS state or get endpoint info from HNS we will need it to covert it back to the `:` format

Reply:
Paul mentioned this is required for delegated vmnics (which is a new nic type-- for infra nics things stay the same)

Reply:
I see. When you say that it is delegated nics does that mean backend nic as well ? If it is only for delegated nic type, should the condition be for delegated nic and not on `all except infranic` ?

Reply:
Hi @vipul-21  this will be applied on delegatedNIC, IBNIC and Accelnet NIC together, I am not sure why they changed the macAddress format but I will follow up with HNS and containerd team.

Reply:
@paul hmm for IB nic, cni not going to invoke hns endpoint creation.. why its required for ib nic?

Reply:
@tamilmani1989 sorry you are right, IB NIC does not interact with HNS endpoint creation. Accelnet NIC does.